internal/pkg/pct/news.go

```diff
@@ -0,0 +1,89 @@
+package pct
+
+import (
+	"bytes"
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/fatih/color"
+	"github.com/rodaine/table"
+)
+
+// The top level structure
+type RSS struct {
+	Item []Item `xml:"channel>item" json:"news"`
+}
+
+// The structure of the information we wish to retrieve
+type Item struct {
+	Title string `xml:"title" json:"title"`
+	Link  string `xml:"link" json:"link"`
+}
+
+func News(Url string, Format string) error {
+	fmt.Printf("Retrieving HTML code of %s ...\n", Url)
+	// Retrieve the html of the page
+	resp, err := http.Get(Url) //#nosec G107
+	if err != nil {
+		return err
+	}
+	// do this now so it won't be forgotten
+	defer resp.Body.Close()
+	// reads html as a slice of bytes
+	html, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	// Create a variable of the top structure type set out above
+	var items RSS
+
+	// Unmarshal the xml
+	err = xml.Unmarshal([]byte(html), &items)
+	if err != nil {
+		return fmt.Errorf("Error ith retrieved yaml")
+	}
+
+	if Format == "table" {
+		outputAsTable(items)
+	} else if Format == "json" {
+		_ = outputAsJson(items)
+	}
+
+	return nil
+}
+
+func outputAsTable(data RSS) {
+	// Output the retrieved results as part of a table
+	// Set the tables format
+	headerFmt := color.New(color.FgGreen, color.Underline).SprintfFunc()
+	columnFmt := color.New(color.FgYellow).SprintfFunc()
+
+	// Add the titles to the table
+	tbl := table.New("Title", "Link")
+	tbl.WithHeaderFormatter(headerFmt).WithFirstColumnFormatter(columnFmt)
+
+	// Add each unmarshalled item as a new row
+	for i := 0; i < len(data.Item); i++ {
+		tbl.AddRow(data.Item[i].Title, data.Item[i].Link)
+	}
```

Comment:
This can be simplified by using `range` https://gobyexample.com/range

Reply:
Updated to:
```
for i := range(data.Item) {
    tbl.AddRow(data.Item[i].Title, data.Item[i].Link)
}
```

Reply:
Range can return the value as well as the index. e.g.

```
for index, value:= range data.Item {
   tbl.AddRow(value.Title, value.Link)
}
```

As you can see with range you can skip the selection of `data.Item[i]` as the value of it is returned by range. In the above example you do not need to declare index, as it's not used, however i've written it there to show what range returns.